File: 7. Arrays/2. Medium/1. 2SumProblem/solution.js
const twoSum = function (nums, target) {
  let pagesMap = {};
  let remainingPages = 0;
  for (let i = 0; i < nums.length; i++) {
    remainingPages = target - nums[i];
    pagesMap[remainingPages] = i;
  }
  for (let i = 0; i < nums.length; i++) {
    if (pagesMap[nums[i]] !== undefined && i !== pagesMap[nums[i]]) {
      return [pagesMap[nums[i]], i];
    }
  }
  return null;
};

File: 7. Arrays/2. Medium/1. 2SumProblem/solution.py
def read(n: int, book: [int], target: int) -> str:
    pages_map = {}
    for pages in book:
        remaining_pages = target - pages
        pages_map[remaining_pages] = pages
    for pages in book:
        if pages in pages_map:
            return "YES"
    return "NO"
